portfolio/src/main/webapp/js/comments.js

```diff
@@ -7,6 +7,14 @@ function initCommentsSystem() {
   const commentForm = document.getElementById("comment-form");
   commentForm.addEventListener("submit", postComment);
 
+  const commentControl = document.getElementById("comment-control");
+  commentControl.addEventListener("change", loadComments);
+
+  commentControl.elements["refresh"].addEventListener("click", (event) => {
+    event.preventDefault();
+    loadComments();
+  });
+
```

Comment:
If possible use HTML attributes to define which javascript functions elements call when events are triggered. For example, you could use "onsubmit=postComment" in the "comment-form" element (https://www.w3schools.com/jsref/event_onsubmit.asp). This is better because:
 1. There is a lot less code. If you set the event listener through javascript you need one line to find the element and 1 line to set the event listener. If you set the event listener through an HTML attribute you just need to add "onsubmit=postComment" which takes up about half a line.
 2. The code I need to read to understand what the element does is in fewer places (which makes it easier to find). If you use an HTML attribute like "onsubmit=postComment" I have to look in 2 places, the HTML element and the postComment function. However, if the event listener is set through javascript I have to look in 3 places: the element, the place where the postComment is set as the event listener, and finally postComment.

Reply:
I see what you're saying, although I have been following the idea of [this](https://google.github.io/styleguide/htmlcssguide.html#Separation_of_Concerns) section of the style guide for HTML that says:
>Strictly keep structure (markup), presentation (styling), and behavior (scripting) apart, and try to keep the interaction between the three to an absolute minimum.
That is, make sure documents and templates contain only HTML and HTML that is solely serving structural purposes. Move everything presentational into style sheets, and everything behavioral into scripts.
In addition, keep the contact area as small as possible by linking as few style sheets and scripts as possible from documents and templates.
Separating structure from presentation from behavior is important for maintenance reasons. It is always more expensive to change HTML documents and templates than it is to update style sheets and scripts.

What do you think? Maybe a good middle ground would be adding some HTML comments above the comment structure that points to the behavior in the comments.js script?

Reply:
Actually I think you're 100% right and you don't need a middle ground. I looked at the HTML/javascript for google.com and it looks like they do everything through addEventListener, not through HTML attributes.